src/posix/platform/radio_spinel.hpp

```diff
@@ -643,7 +650,13 @@ class RadioSpinel : public SpinelInterface::Callbacks
 
     otInstance *mInstance;
 
-    HdlcInterface mHdlcInterface;
+#if OPENTHREAD_POSIX_NCP_UART_ENABLE
```

Comment:
If SPI or UART is determined at compile time, is it still necessary to use virtual methods which is for runtime polymorphism?

Reply:
The Virtual methods indicate which interfaces will be used by `RadioSpinel`. The lower layer only needs to implement these methods.

Reply:
Overall this PR looks great and I am ok with it.

About the virtual use, I do agree with Yakun that the `virtual` methods are not really needed here since we don't need/use runtime polymorphism. As a suggestion for a possible future enhancement, we can consider using CRTP style for `SpinelInterface`. We would then have build time selection option and clear indication of which methods are used/needed by the `RadioSpinel` and full flexibility of having different implementation of interfaces and having no run-time overhead from `virtual` use.

Thoughts?

Reply:
Agree. I will update it.

Reply:
Updated.